### PR TITLE
PIM-9989: Add possibility to clone a client with another index

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

See https://github.com/akeneo/pim-enterprise-dev/pull/11423

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/Infrastructure/PublicApi/IndexMigrationIsDone.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class IndexMigrationIsDone implements IndexMigrationIsDoneInterface
+{
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function byIndexAliasAndHash(string $indexAlias, string $hash): bool
```
Comment (rfauglas) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/Infrastructure/PublicApi/IndexMigrationIsDone.php:29]: Missing integration tests....

**Review by rfauglas — APPROVED** [review]
Great job!
Many improvals.
Further recommendations:
- Do we want to protect against concurrent migration? (According to "Murphy's Law" i'd say yes...!)
- Maybe more utiliity functions in code to clarify intentions.
- Maybe some factory methods so that we can focus on the covered business case in the test...
-  Maybe I would have Introduced an es adaptation layer so that HTTP doesn't leak to the migration service... This way you just have to test integration of individual ES calls in this layer. In the migration service you just validate a proper sequence of service calls in   unit tests... this way you don't really need a full complicated Integration tests... because "integration"  are then turned into "assembly tests ", which are performed by Symfony...

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMappingWithoutDowntime.php [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                        ],
+                    ],
+                    [
+                        'remove' => [
+                            'alias' => $oldIndexAlias,
+                            'index' => $oldIndexName,
+                        ],
+                    ],
+                    [
+                        'add' => [
```
[the remark this answers is not in the document]
Comment (StevenVAIDIE) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMappingWithoutDowntime.php]: Reindexing and switching the index is not atomic. As we didn't have downtime user are allowed to edit / create document during the migration, we reindex documents updated during the index switch

**Review by BitOne — APPROVED** [review]

**Comment by StevenVAIDIE** [comment]
Concurrent migration will be handle by blackhawks

**Pull request merged**